sdk/database/dbplugin/v5/testing/test_helpers.go

```diff
@@ -22,19 +22,46 @@ func getRequestTimeout(t *testing.T) time.Duration {
 	return dur
 }
 
-func AssertInitialize(t *testing.T, db dbplugin.Database, req dbplugin.InitializeRequest) dbplugin.InitializeResponse {
-	t.Helper()
+// AssertInitializeCircleCiTest help to diagnose CircleCI failures within AssertInitialize for mssql tests failing
+// with "Failed to initialize: error verifying connection ..."
+func AssertInitializeCircleCiTest(t *testing.T, db dbplugin.Database, req dbplugin.InitializeRequest) dbplugin.InitializeResponse {
+	maxAttempts := 5
+	var resp dbplugin.InitializeResponse
+	var err error
+	for i := 1; i <= maxAttempts; i++ {
+		resp, err = verifyInitialize(t, db, req)
+		if err != nil {
+			t.Logf("Failed AssertInitialize attempt: %d with error:\n%+v\n", i, err)
```

Comment:
Slightly more idiomatic to use `continue` here, or a `switch` statement,  instead of `else`.  This is fine though.